scripts/keys.coffee

```diff
@@ -22,81 +22,213 @@
 
 module.exports = (robot)->
 
-	key = ()->
-		Key = robot.brain.get("key") or ""
-		robot.brain.set("key" ,Key)
+	kb_key = ()->
+		Key = robot.brain.get("kb_key") or ""
+		robot.brain.set("kb_key" ,Key)
 		Key	
 
-	
-	robot.respond /i have (a key|the key|key|keys)/i, (msg)->
+	ravindra_key = ()->
+		Key = robot.brain.get("ravindra_key") or ""
+		robot.brain.set("ravindra_key" ,Key)
+		Key
+
+	master_key = ()->
+		Key = robot.brain.get("master_key") or ""
+		robot.brain.set("master_key" ,Key)
+		Key
+
+
+#this section belongs to "i have keys"
+	robot.respond /i have (.+) (a key|the key|key|keys)/i, (msg)->
+		keyname = msg.match[1]
 		name = msg.message.user.name 
 		user = robot.brain.userForName name
-		k = key()
-		if typeof user is 'object'
-			msg.send "Okay #{name} has keys"
-			k = user.name
-		robot.brain.set("key",k)	
-
+		if keyname is 'ravindra' or 'rv'
+			k = ravindra_key()
+			if typeof user is 'object'
+				msg.send "Okay #{name} has #{keyname} keys"
```

Comment:
See how you can optimize these if statements. This code segment...
```coffee
msg.send "Okay #{name} has #{keyname} keys"
  robot.brain.set("key",k)						
  k = user.name
  robot.brain.set("ravindra_key",k)
```
...is being repeated 3 times with very little changes